adxl345/adxl345.go

```diff
@@ -0,0 +1,191 @@
+// Package  provides a driver for the digital accelerometer ADXL345
+//
+// Datasheet EN: http://www.analog.com/media/en/technical-documentation/data-sheets/ADXL345.pdf
+// Datasheet JP: http://www.analog.com/media/jp/technical-documentation/data-sheets/ADXL345_jp.pdf
+package adxl345
+
+import (
+	"machine"
+)
+
+type Range uint8
+type Rate uint8
+
+// Internal structure for the power configuration
+type powerCtl struct {
+	link      uint8
+	autoSleep uint8
+	measure   uint8
+	sleep     uint8
+	wakeUp    uint8
+}
+
+// Internal structure for the sensor's data format configuration
+type dataFormat struct {
+	selfTest    uint8
+	spi         uint8
+	intInvert   uint8
+	fullRes     uint8
+	justify     uint8
+	sensorRange Range
+}
+
+// Internal structure for the sampling rate configuration
+type bwRate struct {
+	lowPower uint8
+	rate     Rate
+}
+
+// Device wraps an I2C connection to a BMP180 device.
+type Device struct {
+	bus              machine.I2C
+	Address          uint16
+	powerCtl         powerCtl
+	dataFormat       dataFormat
+	bwRate           bwRate
+	x, y, z          int32
+	rawX, rawY, rawZ int32
+}
+
+// New creates a new BMP180 connection. The I2C bus must already be
+// configured.
+//
+// This function only creates the Device object, it does not touch the device.
+func New(bus machine.I2C, address uint16) Device {
+	return Device{
+		bus: bus,
+		powerCtl: powerCtl{
+			measure: 1,
+		},
+		dataFormat: dataFormat{
+			sensorRange: RANGE_2G,
+		},
+		bwRate: bwRate{
+			lowPower: 1,
+			rate:     RATE_100HZ,
+		},
+		Address: address,
```

Comment:
This should use the default address here, and then the user can set it to some other value if they so desire.